drivers/flash/Kconfig.mcux

```diff
@@ -130,3 +130,32 @@ config FLASH_MCUX_XSPI
 	select FLASH_HAS_EXPLICIT_ERASE
 	help
 	  Enable the mcux xspi flash driver.
+
+if DT_HAS_NXP_S32K3X_C40_ENABLED
```

Comment:
remove and use instead `depends on DT_HAS_NXP_C40_ENABLED`. Note that I intentionally dropped the `S32K3X` from the name beacuse we should aim to have a single DT binding for NXP C40.